browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,457 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_cosmetic_resources_tab_helper.h"
+
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "base/no_destructor.h"
+#include "base/strings/stringprintf.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "brave/components/cosmetic_filters/resources/grit/cosmetic_filters_generated_map.h"
+#include "brave/content/browser/cosmetic_filters_communication_impl.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/chrome_isolated_world_ids.h"
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/global_routing_id.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+#include "ui/base/resource/resource_bundle.h"
+
+namespace {
+
+static base::NoDestructor<std::string> g_observing_script("");
+
+static std::vector<std::string> g_vetted_search_engines = {
+    "duckduckgo", "qwant", "bing", "startpage", "google", "yandex", "ecosia"};
+
+const char kPreInitScript[] =
```

Comment:
these would be better packaged as script resources in grd I think

Reply:
Yeah I suggested that there's no reason and lots to go wrong putting the longer functions in c++ strings vs keeping them as functions in the existing JS script and just calling those functions with the data from here. I think @SergeyZhukovsky preferred to do that as a follow-up and made an issue.